src/goLanguageServer.ts

```diff
@@ -507,13 +507,26 @@ async function latestGopls(tool: Tool): Promise<semver.SemVer> {
 	// Coerce the versions into SemVers so that they can be sorted correctly.
 	const versions = [];
 	for (const version of data.trim().split('\n')) {
-		versions.push(semver.coerce(version));
+		const parsed = semver.parse(version, {
+			includePrerelease: true,
+			loose: true,
+		});
+		versions.push(parsed);
 	}
 	if (versions.length === 0) {
 		return null;
 	}
 	versions.sort(semver.rcompare);
-	return versions[0];
+
+	let latestNonPreRelease: semver.SemVer;
+	for (const version of versions) {
+		// The first version in the sorted list without a prerelease tag.
+		if (version.prerelease.length === 0) {
```

Comment:
Is there a case where `version.prerelease` could be null/undefined or is it always initialized to an empty array?

Reply:
I've only seen it be initialized to an empty array in the cases I've tested.